op-monitorism/secrets/monitor.go

```diff
@@ -0,0 +1,261 @@
+}
+
+func NewMonitor(ctx context.Context, log log.Logger, m metrics.Factory, cfg CLIConfig) (*Monitor, error) {
+	log.Info("creating withdrawals monitor...")
+
+	l1Client, err := ethclient.Dial(cfg.L1NodeURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to dial l1: %w", err)
+	}
+
+	drippie, err := bindings.NewDrippie(cfg.DrippieAddress, l1Client)
+	if err != nil {
+		return nil, fmt.Errorf("failed to bind to Drippie: %w", err)
+	}
+
+	return &Monitor{
+		log: log,
+
+		l1Client: l1Client,
+
+		drippieAddress: cfg.DrippieAddress,
+		drippie:        drippie,
+
+		maxBlockRange: cfg.EventBlockRange,
+		nextL1Height:  cfg.StartingL1BlockHeight,
+
+		// Metrics
+		highestBlockNumber: m.NewGaugeVec(prometheus.GaugeOpts{
+			Namespace: MetricsNamespace,
+			Name:      "highestBlockNumber",
+			Help:      "observed l1 heights (checked and known)",
+		}, []string{"type"}),
+		revealedSecrets: m.NewGaugeVec(prometheus.GaugeOpts{
+			Namespace: MetricsNamespace,
+			Name:      "revealedSecrets",
+			Help:      "revealed secrets",
+		}, []string{"hash"}),
+		nodeConnectionFailures: m.NewCounterVec(prometheus.CounterOpts{
+			Namespace: MetricsNamespace,
+			Name:      "nodeConnectionFailures",
+			Help:      "number of times node connection has failed",
+		}, []string{"layer", "section"}),
+	}, nil
+}
+
+func (m *Monitor) Run(ctx context.Context) {
+	// Determine current L1 block height.
+	latestL1Height, err := m.l1Client.BlockNumber(ctx)
+	if err != nil {
+		m.log.Error("failed to query latest block number", "err", err)
+		m.nodeConnectionFailures.WithLabelValues("l1", "blockNumber").Inc()
+		return
+	}
+
+	// Update metrics.
+	m.highestBlockNumber.WithLabelValues("known").Set(float64(latestL1Height))
+
+	// Find start of the query range.
+	fromBlockNumber := m.nextL1Height
+	if fromBlockNumber > latestL1Height {
+		m.log.Info("no new blocks", "next_height", fromBlockNumber, "latest_height", latestL1Height)
+		return
+	}
+
+	// Find end of the query range.
+	toBlockNumber := latestL1Height
+	if toBlockNumber-fromBlockNumber > m.maxBlockRange {
+		toBlockNumber = fromBlockNumber + m.maxBlockRange
+	}
+
+	// Query for events creating new drips.
+	m.log.Info("querying block range", "from_height", fromBlockNumber, "to_height", toBlockNumber)
+	filterQuery := ethereum.FilterQuery{
+		FromBlock: big.NewInt(int64(fromBlockNumber)),
+		ToBlock:   big.NewInt(int64(toBlockNumber)),
+		Addresses: []common.Address{m.drippieAddress},
+		Topics:    [][]common.Hash{{DripCreatedEventABIHash}},
+	}
+	dripCreatedLogs, err := m.l1Client.FilterLogs(ctx, filterQuery)
+	if err != nil {
+		m.log.Error("failed to query drip created event logs", "err", err)
+		m.nodeConnectionFailures.WithLabelValues("l1", "filterLogs").Inc()
+		return
+	}
+
+	// use this to filter for DripCreated events query
+	m.drippie.DrippieFilterer.FilterDripCreated(
+		&bind.FilterOpts{
+			Start: fromBlockNumber,
+			End:   &toBlockNumber,
+		},
+		nil,
+	)
```

Comment:
Does this actually do anything? We aren't using any return value.

Reply:
Oops, leftover code